src/DependencyInjection/Compiler/HttpCachePass.php

```diff
@@ -22,39 +26,34 @@ class HttpCachePass implements CompilerPassInterface
     public function process(ContainerBuilder $container)
     {
         $this->processHttpDispatcher($container);
-        $this->processVarnishProxyClient($container);
     }
 
     private function processHttpDispatcher(ContainerBuilder $container)
-    {
-        if (!$container->hasDefinition('fos_http_cache.proxy_client.varnish.http_dispatcher')) {
-            return;
-        }
-
-        //Remove FOS default httpDispatcher definition as it fails at validating when using dynamicSettings.
-        //Use \EzSystems\PlatformHttpCacheBundle\ProxyClient\HttpDispatcherFactory instead.
-        $container->removeDefinition('fos_http_cache.proxy_client.varnish.http_dispatcher');
-    }
-
-    private function processVarnishProxyClient(ContainerBuilder $container)
     {
         if (!$container->hasDefinition('fos_http_cache.proxy_client.varnish')) {
             throw new InvalidArgumentException('Varnish proxy client must be enabled in FOSHttpCacheBundle');
         }
+        // Override FOS default httpDispatcher with \EzSystems\PlatformHttpCacheBundle\ProxyClient\HttpDispatcherFactory.
+        $container->removeDefinition('fos_http_cache.proxy_client.varnish.http_dispatcher');
 
         $fosConfig = array_merge(...$container->getExtensionConfig('fos_http_cache'));
 
         $servers = $fosConfig['proxy_client']['varnish']['http']['servers'] ?? [];
         $baseUrl = $fosConfig['proxy_client']['varnish']['http']['base_url'] ?? '';
```

Comment:
Do we have these exposed as parameters? If not, then ok, decorating might not be possible in a clean way.